applications/DigitalTwinApplication/custom_sensors/measurement_residual_response_function.cpp

```diff
@@ -0,0 +1,296 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         DigitalTwinApplication/license.txt
+//
+//  Main authors:    Suneth Warnakulasuriya
+//
+
+// System includes
+#include <cmath>
+#include <sstream>
+
+// External includes
+
+// Project includes
+#include "utilities/openmp_utils.h"
+#include "utilities/parallel_utilities.h"
+
+// Application includes
+#include "digital_twin_application_variables.h"
+
+// Include base h
+#include "measurement_residual_response_function.h"
+
+namespace Kratos
+{
+
+namespace MeasurementResidualResponseFunctionUtilities
+{
+
+struct GradientCalculation {
+    template <class TEntityType>
+    static inline void Calculate(
+        Sensor& rSensor,
+        Vector& rResponseGradient,
+        const Matrix& rResidualGradient,
+        const TEntityType& rEntity,
+        const ProcessInfo& rProcessInfo)
+    {
+        rSensor.CalculateGradient(rEntity, rResidualGradient, rResponseGradient, rProcessInfo);
+    }
+};
+
+struct FirstDerivativesGradientCalculation {
+    template <class TEntityType>
+    static inline void Calculate(
+        Sensor& rSensor,
+        Vector& rResponseGradient,
+        const Matrix& rResidualGradient,
+        const TEntityType& rEntity,
+        const ProcessInfo& rProcessInfo)
+    {
+        rSensor.CalculateFirstDerivativesGradient(rEntity, rResidualGradient, rResponseGradient, rProcessInfo);
+    }
+};
+
+struct SecondDerivativesGradientCalculation {
```

Comment:
if we have Hessian we can use Newton Methods.

Reply:
nice :)